contracts/core/trustless/WitOracleRadonRegistryDefault.sol

```diff
@@ -315,39 +318,82 @@ contract WitOracleRadonRegistryDefault
             _reducer.validate();
             __reducer.opcode = _reducer.opcode;
             __pushRadonReducerFilters(__reducer, _reducer.filters);
-            emit NewRadonReducer(hash);
+            emit NewRadonReducer(bytes16(hash));
         }
     }
     
     function verifyRadonRequest(
-            bytes32[] calldata retrieveHashes,
-            Witnet.RadonReducer calldata aggregate,
-            Witnet.RadonReducer calldata tally
+            bytes32[] calldata _retrieveHashes,
+            Witnet.RadonReducer calldata _aggregateReducer,
+            Witnet.RadonReducer calldata _tallyReducer
         ) 
         override external 
         returns (bytes32 radHash)
     {
-        // TODO
+        return __verifyRadonRequest(
+            _retrieveHashes,
+            new string[][](_retrieveHashes.length),
+            _aggregateReducer,
+            _tallyReducer
+        );
+    }
+
+    function verifyRadonRequest(
+            bytes32[] calldata _retrieveHashes,
+            bytes32 _aggregateReducerHash,
+            bytes32 _tallyReducerHash
+        )
+        override external
+        returns (bytes32 radHash)
+    {
+        return __verifyRadonRequest(
+            _retrieveHashes,
+            new string[][](_retrieveHashes.length),
+            _aggregateReducerHash,
+            _tallyReducerHash
+        );
     }
 
     function verifyRadonRequest(
             bytes32[] calldata _retrieveHashes,
-            string[][] calldata _retrieveArgs,
-            Witnet.RadonReducer calldata _aggregate,
-            Witnet.RadonReducer calldata _tally
+            string[][] calldata _retrieveArgsValues,
+            Witnet.RadonReducer calldata _aggregateReducer,
+            Witnet.RadonReducer calldata _tallyReducer
         ) 
         override external 
-        returns (bytes32 _radHash)
+        returns (bytes32)
     {
-        // TODO
+        return __verifyRadonRequest(
+            _retrieveHashes, 
+            _retrieveArgsValues, 
+            _aggregateReducer, 
+            _tallyReducer
+        );
+    }
+
+    function verifyRadonRequest(
+            bytes32[] calldata _retrieveHashes,
+            string[][] calldata _retrieveArgsValues,
+            bytes32 _aggregateReducerHash,
+            bytes32 _tallyReducerHash
+        )
+        override external
+        returns (bytes32)
+    {
+        return __verifyRadonRequest(
+            _retrieveHashes,
+            _retrieveArgsValues,
+            _aggregateReducerHash,
+            _tallyReducerHash
+        );
     }
 
     function verifyRadonRetrieval(
             Witnet.RadonRetrievalMethods _requestMethod,
-            string calldata _requestURL,
-            string calldata _requestBody,
+            string memory _requestURL,
+            string memory _requestBody,
             string[2][] memory  _requestHeaders,
-            bytes calldata _requestRadonScript
+            bytes memory _requestRadonScript
```

Comment:
Is it necessary to replace `calldata` with `memory`? Isn't `calldata` cheaper than `memory`?

Reply:
> Is it necessary to replace `calldata` with `memory` ?
Yep, as `verifyRadonRetrieval(uint8,string,string,string[][],bytes)` is now internally called from the implementation of the (new) `verifyRadonRetrieval(bytes32,string)` which is getting _requestURL, _requestBody, _requestHeaders and `_requestRadonScript` as modified memory buffers as returned from `lookupRadonRetrieval(bytes32)`.

> `Isn't `calldata` cheaper than `memory` ?
It depends. Not necessarily. In some cases using `calldata` parameters may slightly reduce the gas consumption, but not in this case whatsoever, as formerly `calldata` parameters had to be copied to storage anyway.